src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ILScanner.cs

```diff
@@ -543,7 +543,12 @@ public ScannedDevirtualizationManager(NodeFactory factory, ImmutableArray<Depend
                             TypeDesc canonType = type.ConvertToCanonForm(CanonicalFormKind.Specific);
 
                             if (!canonType.IsDefType || !((MetadataType)canonType).IsAbstract)
+                            {
                                 _canonConstructedTypes.Add(canonType.GetClosestDefType());
+                                TypeDesc canonBaseType = canonType.BaseType?.ConvertToCanonForm(CanonicalFormKind.Specific);
+                                while (canonBaseType != null && _canonConstructedTypes.Add(canonBaseType))
+                                    canonBaseType = canonType.BaseType?.ConvertToCanonForm(CanonicalFormKind.Specific);
```

Comment:
```suggestion
                                DefType closestDefCanonType = canonType.GetClosestDefType();
                                _canonConstructedTypes.Add(closestDefCanonType);
                                
                                TypeDest baseType = closestDefCanonType.BaseType;
                                while (baseType != null)
                                {
                                    baseType = baseType.ConvertToCanonForm(CanonicalFormKind.Specific);
                                    if (_canonConstructedTypes.Add(baseType))
                                       break;
                                    baseType = baseType.BaseType;
                                }
```
This is same loop as below. It would be nice to follow the same style for both loops. (If you take my suggestion, replace `bool added = true` with a `break;`.)